### PR TITLE
docs: add lambda pre-aggs and ksqldb

### PR DESCRIPTION
**Pull request by keydunov**

### CODE REVIEW

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+title: Lambda Pre-Aggregations
+permalink: /caching/pre-aggregations/lambda-pre-aggregations
+category: Caching
+menuOrder: 4
+---
+
+Lambda pre-aggregations follow the [Lambda architecture](https://en.wikipedia.org/wiki/Lambda_architecture) design to union real-time and batch data. Cube acts as a serving layer and uses pre-aggregations as a batch layer and source data or other pre-aggregations, usually [streaming][streaming-pre-agg], as a speed layer.
```
Comment (rpaik) [inline docs/content/Caching/Lambda-Pre-Aggregations.mdx:8]: Would it be better to say "service layer" (vs. "serving layer")?

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+<WarningBox>
+
+Lambda pre-aggregations only work with Cube Store. 
+
+Additionally, we’re going to remove support for external storages, other than Cube Store, later this year. Cube Store will replace Redis and, therefore will be a required component to run Cube even without pre-aggregations.
```
Comment (rpaik) [inline docs/content/Caching/Lambda-Pre-Aggregations.mdx:14]:
```suggestion
Additionally, we’re going to remove support for external storages, other than Cube Store, later this year. [Cube Store will replace Redis](https://cube.dev/blog/replacing-redis-with-cube-store) and, therefore will be a required component to run Cube even without pre-aggregations.
```

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  />
+</div>
+
+First, you need to create pre-aggregations that will contain your batch data. In the following example, we call it **batch.** Please note, it must have `timeDimension`, Cube will use it to union batch data with source data.
+
+You control the batch part of your data with `buildRangeStart` and `buildRangeEnd` properties of pre-aggregation to determine specific window for your batched data.
+
+Next, you need to create a lambda pre-aggregation. To do that, create pre-aggregation with type `rollupLambda`, specify rollups you would like to use with `rollups` property, and finally set `unionWithSourceData: true` to use source data as a real-time layer.
+
+Please make sure that lambda pre-aggreation definition goes first when defining your pre-aggregations.
```
Comment (rpaik) [inline docs/content/Caching/Lambda-Pre-Aggregations.mdx:41]:
```suggestion
Please make sure that the lambda pre-aggreation definition comes first when defining your pre-aggregations.
```

File: docs/content/Caching/Using-Pre-Aggregations.mdx [hunk 4]
```diff
@@ -479,6 +479,22 @@ When using cloud storage, it is important to correctly configure any data
...
 
+## Streaming pre-aggregations
+
+Streaming pre-aggregations are different from traditional pre-aggregations in the way they are being updated. Traditional pre-aggregations follow the “pull” model — Cube **pulls updates** from the data source based on some cadence and/or condition. Streaming pre-aggregations follow the “push” model — Cube **subscribes to the updates** from the data source and always keeps pre-aggregation up to date. 
+
+You don’t need to define `refreshKey` for streaming pre-aggregations. Whether pre-aggregation is streaming or not is defined by the data source. 
+
+Currently, Cube supports only one streaming data source - [ksqlDB](/config/databases/ksqldb). All pre-aggregations where data source is ksqlDB are streaming. 
+
+We’re working to support streaming pre-aggregations for the following data sources -
```
Comment (rpaik) [inline docs/content/Caching/Using-Pre-Aggregations.mdx:490]:
```suggestion
We’re working on supporting streaming pre-aggregations for the following data sources -
```

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx (hunk 1 above)
Comment (keydunov) [reply docs/content/Caching/Lambda-Pre-Aggregations.mdx:8]: It is specific term used in the lambda architecture - https://en.wikipedia.org/wiki/Lambda_architecture

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  />
+</div>
+
+First, you need to create pre-aggregations that will contain your batch data. In the following example, we call it **batch.** Please note, it must have `timeDimension`, and Cube will use it to union batch data with source data.
+
+You control the batch part of your data with `buildRangeStart` and `buildRangeEnd` properties of pre-aggregation to determine specific window for your batched data.
+
+Next, you need to create a lambda pre-aggregation. To do that, create pre-aggregation with type `rollupLambda`, specify rollups you would like to use with `rollups` property, and finally set `unionWithSourceData: true` to use source data as a real-time layer.
+
+Please make sure that the lambda pre-aggreation definition comes first when defining your pre-aggregations.
```
Comment (hassankhan) [inline docs/content/Caching/Lambda-Pre-Aggregations.mdx:41]:
```suggestion
Please make sure that the lambda pre-aggregation definition comes first when defining your pre-aggregations.

```

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+</WarningBox>
+
+## Use cases
+
+Below we are looking at the most common examples of using lambda pre-aggregations.
+
+### Batch and source data
+
+Batch data is coming from pre-aggregation and real-time data is coming from the data source.
```
Comment (hassankhan) [inline docs/content/Caching/Lambda-Pre-Aggregations.mdx:24]:
```suggestion
In this scenario, batch data comes from a pre-aggregation and real-time data comes from the data source.
```

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  },
+  buildRangeEnd: {
+    sql: `SELECT '2022-05-30'`
+  }
+}
+```
+
+### Batch and streaming data
+
+Batch data is coming from one pre-aggregation and real-time data coming from another, [streaming pre-aggregation][streaming-pre-agg].
```
Comment (hassankhan) [inline docs/content/Caching/Lambda-Pre-Aggregations.mdx:65]:
```suggestion
In this scenario, batch data is comes from one pre-aggregation and real-time data comes from a [streaming pre-aggregation][streaming-pre-agg].
```

File: docs/content/Caching/Lambda-Pre-Aggregations.mdx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    style="border: none"
+    width="100%"
+  />
+</div>
+
+
+You can use lambda pre-aggregations to combine data from multiple pre-aggregation, where one pre-aggregation can have batch data and another streaming.
+
+```js
+batchStreamingLambda: {
```
Comment (hassankhan) [inline docs/content/Caching/Lambda-Pre-Aggregations.mdx:80]: Could we add the `cube()` definition for this example? It looks like the example uses two cubes, but its not clear which one the pre-aggregation definition lives in

Comment (keydunov) [reply docs/content/Caching/Lambda-Pre-Aggregations.mdx:81]: Yes, let's do it! It is a good idea. Can you please send it as a separate PR?

**Pull request merged**